src/order/complete_lattice.lean

```diff
@@ -1041,12 +1041,20 @@ eq_of_forall_le_iff $ λ c, by simp only [le_infi_iff, sigma.forall]
 theorem supr_sigma {p : β → Type*} {f : sigma p → α} : (⨆ x, f x) = (⨆ i (h : p i), f ⟨i, h⟩) :=
 @infi_sigma αᵒᵈ _ _ _ _
 
-theorem infi_prod {γ : Type*} {f : β × γ → α} : (⨅ x, f x) = (⨅ i j, f (i, j)) :=
+theorem infi_prod {f : β × γ → α} : (⨅ x, f x) = (⨅ i j, f (i, j)) :=
 eq_of_forall_le_iff $ λ c, by simp only [le_infi_iff, prod.forall]
 
-theorem supr_prod {γ : Type*} {f : β × γ → α} : (⨆ x, f x) = (⨆ i j, f (i, j)) :=
+theorem supr_prod {f : β × γ → α} : (⨆ x, f x) = (⨆ i j, f (i, j)) :=
 @infi_prod αᵒᵈ _ _ _ _
 
+lemma bsupr_prod {f : β × γ → α} {s : set β} {t : set γ} :
+  (⨆ x ∈ s ×ˢ t, f x) = ⨆ (a ∈ s) (b ∈ t), f (a, b) :=
```

Comment:
Do you definitely want the binders in this order and not the following?
```suggestion
  (⨆ x ∈ s ×ˢ t, f x) = ⨆ a b (ha : a ∈ s) (hb : b ∈ t), f (a, b) :=
```
(I'm not saying that changing it is a good idea, I just want to check if this is deliberate)

Reply:
Yes. I had been using `a b ha hb` for a very long time but I have now changed my mind to use `a ha b hb` instead because this allows you to manipulate `a ha` as a single entity. For example by rewriting `set.Union₂_comm` or using the fact that `s.bUnion f = ⋃ x ∈ ↑s, f x` for `s` a finset.